implementations/rust/ockam/ockam_command/src/node/delete.rs

```diff
@@ -31,94 +29,38 @@ impl DeleteCommand {
 }
 
 fn run_impl(opts: CommandGlobalOpts, cmd: DeleteCommand) -> crate::Result<()> {
-    if cmd.all {
-        // Try to delete all nodes found in the config file + their associated processes
-        let nn: Vec<String> = {
-            let inner = &opts.config.inner();
-            inner.nodes.iter().map(|(name, _)| name.clone()).collect()
-        };
-        for node_name in nn.iter() {
-            delete_node(&opts, node_name, cmd.force)
-        }
-
-        // Try to delete dangling embedded nodes directories
-        let dirs = OckamConfig::directories();
-        let nodes_dir = dirs.data_local_dir();
-        for entry in nodes_dir.read_dir()? {
-            let dir = entry?;
-            if !dir.file_type()?.is_dir() {
-                continue;
-            }
-            if let Some(dir_name) = dir.file_name().to_str() {
-                if !nn.contains(&dir_name.to_string()) {
-                    let _ = std::fs::remove_dir_all(dir.path());
+    let res = delete_nodes_with_options(opts.clone(), cmd.all, cmd.force);
+    /*
+       Previous impl:
+            if cmd.all {
+                //reducted
+                if cmd.force {
+                    // reducted
+                } else if let Err(e) = opts.config.persist_config_updates() {
+                    // reducted
                 }
+            } else {
+                // reducted
             }
-        }
 
-        // If force is enabled
-        if cmd.force {
-            // delete the config and nodes directories
-            opts.config.remove()?;
-            // and all dangling/orphan ockam processes
-            if let Ok(cpid) = get_current_pid() {
-                let s = System::new_all();
-                for (pid, process) in s.processes() {
-                    if pid != &cpid && process.name() == "ockam" {
-                        process.kill();
-                    }
-                }
+       If Both --All and --Force were set, `persist_config_updates` is not needed
+       If --All was set and not --Force, we have to call `persist_config_updates`
+       If --Force was set and not --All, we fall back to this else condition with `delete_node`
+       If None, we fall back to this else condition with `delete_node`
+    */
+    if cmd.all && res.is_ok() {
+        // If force is not enabled
+        if !cmd.force {
+            // persist updates to the config file
+            if let Err(e) = opts.config.persist_config_updates() {
+                eprintln!("Failed to update config file. You might need to run the command with --force to delete all config directories");
+                return Err(crate::Error::new(crate::exitcode::IOERR, e));
```

Comment:
Why not keep this inside the `delete_nodes_with_options` function?

Reply:
moved it to `delete_all_nodes()`